### PR TITLE
issue #943 change isExportPublic etc to tenant specific

### PR DESCRIPTION
**Pull request by albertwang-ibm**
Changed these following bulkdata configurations to tenant specific:
    - fhirServer/bulkdata/isExportPublic
    - fhirServer/bulkdata/cosFileMaxResources
    - fhirServer/bulkdata/cosFileMaxSize

Signed-off-by: Albert Wang <xuwang@us.ibm.com>

### CODE REVIEW

File: fhir-bulkimportexport-webapp/src/main/java/com/ibm/fhir/bulkexport/system/ChunkWriter.java [hunk 1]
```diff
@@ -200,4 +201,11 @@ public void writeItems(List<java.lang.Object> arg0) throws Exception {
             }
         }
     }
+
+    @Override
+    public void open(Serializable checkpoint) throws Exception
+    {
+        isExportPublic = FHIRConfigHelper.getBooleanProperty(FHIRConfiguration.PROPERTY_BULKDATA_BATCHJOB_ISEXPORTPUBLIC, true);
```
Comment (prb112) [inline fhir-bulkimportexport-webapp/src/main/java/com/ibm/fhir/bulkexport/system/ChunkWriter.java:208]: WE could pass this in as a part of the job parameters

File: fhir-bulkimportexport-webapp/src/main/java/com/ibm/fhir/bulkexport/system/ChunkWriter.java [hunk 2]
```diff
@@ -200,4 +201,11 @@ public void writeItems(List<java.lang.Object> arg0) throws Exception {
             }
         }
     }
+
+    @Override
+    public void open(Serializable checkpoint) throws Exception
+    {
```
Comment (prb112) [inline fhir-bulkimportexport-webapp/src/main/java/com/ibm/fhir/bulkexport/system/ChunkWriter.java:207]:
```suggestion
    public void open(Serializable checkpoint) throws Exception  {
```

File: fhir-bulkimportexport-webapp/src/main/java/com/ibm/fhir/bulkexport/common/CheckPointAlgorithm.java [hunk 3]
```diff
@@ -53,38 +51,28 @@ public CheckPointAlgorithm() {
...
-    /**
-     * @see CheckpointAlgorithm#isReadyToCheckpoint()
-     */
     @Override
     public boolean isReadyToCheckpoint() {
         TransientUserData chunkData = (TransientUserData) stepCtx.getTransientUserData();
 
         if (chunkData != null) {
+            int cosFileMaxResources = FHIRConfigHelper.getIntProperty(FHIRConfiguration.PROPERTY_BULKDATA_BATCHJOB_COSFILEMAXRESOURCES, Constants.DEFAULT_COSFILE_MAX_RESOURCESNUMBER);
+            int cosFileMaxSize = FHIRConfigHelper.getIntProperty(FHIRConfiguration.PROPERTY_BULKDATA_BATCHJOB_COSFILEMAXSIZE  , Constants.DEFAULT_COSFILE_MAX_SIZE);
```
Comment (prb112) [inline fhir-bulkimportexport-webapp/src/main/java/com/ibm/fhir/bulkexport/common/CheckPointAlgorithm.java:75]:
```suggestion
            int cosFileMaxSize = FHIRConfigHelper.getIntProperty(FHIRConfiguration.PROPERTY_BULKDATA_BATCHJOB_COSFILEMAXSIZE , Constants.DEFAULT_COSFILE_MAX_SIZE);
```

File: fhir-bulkimportexport-webapp/src/main/java/com/ibm/fhir/bulkexport/system/ChunkWriter.java (hunk 1 above)
Comment (albertwang-ibm) [reply fhir-bulkimportexport-webapp/src/main/java/com/ibm/fhir/bulkexport/system/ChunkWriter.java:208]: I thought about this also and then finally wanted to make it simpler to allow the the job pick up the configuration by itself. we already have so many job parameters. :)

File: fhir-bulkimportexport-webapp/src/main/java/com/ibm/fhir/bulkexport/system/ChunkWriter.java (hunk 2 above)
Comment (albertwang-ibm) [reply fhir-bulkimportexport-webapp/src/main/java/com/ibm/fhir/bulkexport/system/ChunkWriter.java:207]: ha, good catch.

**Pull request merged**